### PR TITLE
Embed exporters into agent jar

### LINKED ISSUE #370: Have friendlier export config
Right now to do export you need to download/find the exporter jar and then configure it like
`-Dota.exporter.jar=path/to/opentelemetry-auto-exporters-jaeger-<version>.jar `.  While still keeping the ability to create a custom jar with a custom SpanExporter{Factory}, we could embed the code for our supported exporters into our fat jar, and allow more convenient shorthand config like `-Dota.exporter=jeager`.

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #370 

Publish two versions of java agent: new one, with all available exports embedded into jar file, and old one, without exporters. Version with embedded exporters is a "default" one and have no classifier. Version without exporters has the "no-exporters" classifier. The former is to be used by all clients, the latter is mainly meant for vendors to repackage with their own exporters.

Old way of configuring exporter by providing external jar is still supported without any modifications. New, simpler configuration is to provide `ota.exporter` configuration options with possible values of `otlp`, `logging`, `jaeger` and `zipkin`. If no configuration is provided then `otlp` exporter is used. It is currently impossible to configure different exporters for traces and metrics.

There is now significant size differences between embedding only `otlp` exporter and embedding them all.

Documentation was NOT changed on purpose. I suggest to update documentation right after next release, when new jar is available from the location pointed by our documentation.

### CODE REVIEW

File: auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpMetricExporterFactory.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
   @Override
   public MetricExporter fromConfig(final Config config) {
-    final String otlpEndpoint = config.getString(OTLP_ENDPOINT, "");
+    final String otlpEndpoint = config.getString(OTLP_ENDPOINT, "localhost:55680");
```
[the remark this answers is not in the document]
Comment (iNikem) [reply auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpMetricExporterFactory.java:29]: Took that from Otel collector.

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java [hunk 2]
```diff
@@ -54,7 +60,33 @@ public static synchronized void installAgentTracer() {
...
+        ServiceLoader.load(SpanExporterFactory.class, TracerInstaller.class.getClassLoader());
+
+    for (SpanExporterFactory spanExporterFactory : serviceLoader) {
+      if (spanExporterFactory
+          .getClass()
+          .getSimpleName()
+          .toLowerCase()
+          .startsWith(exporterName.toLowerCase())) {
+        return spanExporterFactory;
+      }
```
[the remark this answers is not in the document]
Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:85]: Yeah, I didn't like it neither. But I could not come up with any other way to identify implementations of `SpanExporterFactory`.

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java [hunk 3]
```diff
@@ -70,33 +102,46 @@ private static synchronized void installExportersFromJar(final String exporterJa
...
-      final MetricExporter metricExporter = metricExporterFactory.fromConfig(config);
-      IntervalMetricReader.builder()
-          .setMetricExporter(metricExporter)
-          .setMetricProducers(
-              Collections.singleton(OpenTelemetrySdk.getMeterProvider().getMetricProducer()))
-          .build();
-      log.info("Installed metric exporter: " + metricExporter.getClass().getName());
+      installExporter(metricExporterFactory, config);
+    } else {
+      log.warn("No metric exporter found in {}. All metrics are dropped", exporterJar);
```
[the remark this answers is not in the document]
Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:118]: What is the plan for metric exporters and their configuration? If I want to send spans to Zipkin and metrics to Prometheus, how should I configure my agent?

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:80]: Maybe it makes sense to open an issue in `otel-java` to add some `id/name` method to `SpanExporterFactory`?

[the remark this answers is not in the document]
Comment (trask) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:80]: Yes, I like that, I forgot that `SpanExporterFactory` is basically just for auto-instrumentation anyways, so no reason not to add stuff in there that we need 👍

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (trask) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: Good question. Can you open an issue to track this?

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java:80]: Then I will leave this as it is now for now and open an issue in `otel-java`. After that is done, will update this condition here in a future PR.

File: agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (iNikem) [reply agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java]: #496

**Comment by iNikem** [comment]
Created new task open-telemetry/opentelemetry-java#1332

**Pull request merged**